adapters/dmx/dmx.go

```diff
@@ -0,0 +1,245 @@
+package dmx
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbs"
+	"net/http"
+	"sort"
+)
+
+type DmxAdapter struct {
+	endpoint    string
+	publisherId string
+}
+
+func New(text string) error {
+	return &errorString{text}
+}
+
+// errorString is a trivial implementation of error.
+type errorString struct {
+	s string
+}
+
+func (e *errorString) Error() string {
+	return e.s
+}
+
+func NewDmxBidder(endpoint string, publisher_id string) *DmxAdapter {
+	return &DmxAdapter{endpoint: endpoint, publisherId: publisher_id}
+}
+
+type dmxBidder struct {
+	Bidder dmxExt `json:"bidder"`
+}
+
+type dmxUser struct {
+	User *openrtb.User `json:"user"`
+}
+
+type dmxExt struct {
+	PublisherId string `json:"publisher_id,omitempty"`
+}
+
+type dmxBanner struct {
+	Banner *openrtb.Banner `json:"banner"`
+}
+
+type dmxSize struct {
+	W uint64
+	H uint64
+	S uint64
+}
+
+type DmxSize []dmxSize
```

Comment:
Consider naming this `DmxSizes` to indicate it's a collection rather than a single unit.